settings.gradle

```diff
@@ -11,3 +11,4 @@ include ':public'
 include ':purchases'
 include 'examples:purchase-tester'
 include ':api-tester'
+include 'examples:MagicWeather:app'
```

Comment:
MagicWeather is meant to be a sample app which doesn't actually use the local code directly but fetches the version from maven, that's why it's configured as a separate project. I believe trying to reuse it in the root project is not a good idea since people might expect that changes in the SDK would affect MagicWeather. Usually I open it on a different Android Studio window. What do you think? 

Also, I'm not sure if this would ignore MagicWeather's project gradle configuration in favor of using the SDK project gradle configuration, which we might not want to do.

Reply:
> MagicWeather is meant to be a sample app which doesn't actually use the local code directly but fetches the version from maven, that's why it's configured as a separate project. I believe trying to reuse it in the root project is not a good idea since people might expect that changes in the SDK would affect MagicWeather. Usually I open it on a different Android Studio window. What do you think?

When you say "people" here, who are they? I mean, who is the target of `MagicWeather`? It seemed to me that it is the end-developers that want to play with a working app and see how the SDK APIs look like in the latest stable version (like myself 😅). If so, yeah you may want to keep it separate from the rest of the modules.

At the same time, as a contributor of `purchases-android`, at a first glance, I'd expect to load the project and be able to run the sample apps that are included in `examples`. But I hear you, after poking around you quickly learn that `purchase-tester` is targeted towards that.

That being said, to avoid any kind of confusion, I think that anyone who land at the project would appreciate if that's clear, so how about moving the `MagicWeather` sample app to a separate repository? I know that doing that may entail more work so another quick short-term solution would be to add a note to the `README` around the lines of what you mentioned above: `Usually I open it on a different Android Studio window`. How does that sound?

> Also, I'm not sure if this would ignore MagicWeather's project gradle configuration in favor of using the SDK project gradle configuration, which we might not want to do.

While 🔨 I didn't run into any issues so it shouldn't be a problem, I'd expect AS / Gradle handling that smoothly 😄

Reply:
Yeah, sorry I should have explained better. We currently have 2 apps in the android sdk repo:
- `purchase-tester`: This is meant to be an internal testing app. It's configured in a way that makes it easier to test all the different SDK functionalities, but might not reflect an actual app using SDK. This is what we mostly use internally. In that sense, `purchase-tester` shouldn't be in the `examples` folder, but probably in a different folder (maybe `test-apps`). That's something we agree it's not ideal in the current state.
- `MagicWeather`: As you said, it's meant to be an actual sample app. It should be what users of our SDK look for when they want to implement the SDK and want to see an example of the most common use cases.

With the change in this PR, developers can still open `MagicWeather` as a standalone project but we (as in RC devs and contributors) will be making any changes through the root SDK project, resulting in a disconnection on how people looking for a sample app open `MagicWeather` and how we do it. I think it's better to keep `MagicWeather` as a standalone app to make sure the experience we get is aligned with what users have. 

> That being said, to avoid any kind of confusion, I think that anyone who land at the project would appreciate if that's clear, so how about moving the MagicWeather sample app to a separate repository?

That's an option, but for now, we were trying to keep everything in the same place to make it easy to access. We've discussed restructuring the repo so it's less confusing though but we haven't gotten around to do it. 

> I know that doing that may entail more work so another quick short-term solution would be to add a note to the README around the lines of what you mentioned above: Usually I open it on a different Android Studio window. How does that sound?

That could be a good short-term compromise until we actually restructure the SDK 👍 

Does that make sense? Lmk if you have other thoughts!

Reply:
Crystal clear ✨

Reply:
> That could be a good short-term compromise until we actually restructure the SDK 👍

Follow-up in https://github.com/RevenueCat/purchases-android/pull/1041